### PR TITLE
Laboratory work #2, Biryukova Ksenia - 19FPL2

### PR DESCRIPTION
**Pull request by biryukovaksenia**

### CODE REVIEW

File: lab_2/main.py [hunk 1]
```diff
@@ -34,7 +46,20 @@ def fill_lcs_matrix(first_sentence_tokens: tuple, second_sentence_tokens: tuple)
...
+    lcs = []
+    rows = len(first_sentence_tokens)
+    columns = len(second_sentence_tokens)
+    if rows <= 0 and columns <= 0:
+        return []
+    if isinstance(first_sentence_tokens, tuple) and isinstance(second_sentence_tokens, tuple):
+        lcs = create_zero_matrix(rows, columns)
+        for ind_1, token_1 in enumerate(first_sentence_tokens):
+            for ind_2, token_2 in enumerate(second_sentence_tokens):
+                if token_1 == token_2:
```
Comment (ValeriyaKuznetsova) [inline lab_2/main.py]: what happens if ind_1 == 0 or ind_2 == 0?

File: lab_2/main.py [hunk 2]
```diff
@@ -81,7 +168,29 @@ def calculate_text_plagiarism_score(original_text_tokens: tuple, suspicious_text
...
+              and isinstance(plagiarism_threshold, float) and 0 <= plagiarism_threshold <= 1
+    score_sum = 0
+
+    while len(suspicious_text_tokens) > len(original_text_tokens):
+        original_text_tokens += ('',)
+    if len(suspicious_text_tokens) < len(original_text_tokens):
+        original_text_tokens = original_text_tokens[:len(suspicious_text_tokens)]
+
+    if is_data:
+        for ind1, token1 in enumerate(original_text_tokens):
```
Comment (ValeriyaKuznetsova) [inline lab_2/main.py]: you need to iterate only over suspicious_text_tokens

File: lab_2/main.py [hunk 3]
```diff
@@ -92,7 +201,21 @@ def find_diff_in_sentence(original_sentence_tokens: tuple, suspicious_sentence_t
...
+              and isinstance(lcs, tuple)
+    if not is_data:
+        return ()
+
+    changed_tokens = []
+    ind_of_changed = []
+
+    for ind, token in enumerate(original_sentence_tokens):
+        if token not in lcs:
+            changed_tokens.append(ind)
```
Comment (ValeriyaKuznetsova) [inline lab_2/main.py]: not all the requirements are satisfied, read the description once more, please

File: lab_2/main.py [hunk 4]
```diff
@@ -46,19 +48,21 @@ def fill_lcs_matrix(first_sentence_tokens: tuple, second_sentence_tokens: tuple)
...
-            for ind_2, token_2 in enumerate(second_sentence_tokens):
-                if token_1 == token_2:
-                    lcs[ind_1][ind_2] = lcs[ind_1 - 1][ind_2 - 1] + 1
-                else:
-                    lcs[ind_1][ind_2] = max(lcs[ind_1 - 1][ind_2], lcs[ind_1][ind_2 - 1])
+
+    lcs = create_zero_matrix(len(first_sentence_tokens),len(second_sentence_tokens))
+    for ind_1, token_1 in enumerate(first_sentence_tokens):
+        for ind_2, token_2 in enumerate(second_sentence_tokens):
+            if token_1 == token_2:
```
Comment (ValeriyaKuznetsova) [inline lab_2/main.py:63]: still, have a question about ind_1 ==0 and index ==0

**Review by ValeriyaKuznetsova — COMMENTED** [review]
waiting for your answer about 0s

update start.py and target_score.txt

File: lab_2/main.py (hunk 4 above)
Comment (biryukovaksenia) [reply lab_2/main.py:63]: if ind_1 == 0 or ind_2 == 0, we're on the last row or colomn and have indexes [-1],  other rows or columns become 1 if token_1 == token_2. And than we have another iteration.

**Pull request closed without merging**